modules/gloo/pkg/vsfactory/vsfactory.go

```diff
@@ -18,10 +21,16 @@ import (
 )
 
 const (
-	rioNameHeader      = "X-Rio-ServiceName"
-	rioNamespaceHeader = "X-Rio-Namespace"
+	rioNameHeader         = "X-Rio-ServiceName"
+	rioNamespaceHeader    = "X-Rio-Namespace"
+	scaleFromZeroAttempts = 10
 )
 
+// Retry configuration for the envoy gateway (Gloo)  https://www.envoyproxy.io/learn/automatic-retries
+// for services scaled to zero
+var overallTimeout = time.Second * 30
+var perTryTimeout = time.Second * 2
```

Comment:
This was what I saw as a decent timeout for each request. Ultimately I think this may need to be configurable based upon the service being scaled